File: src/adapters/claude-code.ts
#!/usr/bin/env node
/** Claude Code adapter — UserPromptSubmit hook.
 *
 * Reads CC's stdin JSON {"prompt":"..."}, classifies, emits hookSpecificOutput.
 *
 * Wire in ~/.claude/settings.json:
 *   "UserPromptSubmit": [{"matcher":"","hooks":[{"type":"command",
 *     "command":"node /path/to/dist/adapters/claude-code.js"}]}]
 */
import { classify, FALLBACK } from "../classify.js";

const raw = await new Promise<string>((res) => {
  let buf = "";
  process.stdin.setEncoding("utf8");
  process.stdin.on("data", (d) => (buf += d));
  process.stdin.on("end", () => res(buf));
});

let prompt = "";
try {
  const parsed = JSON.parse(raw);
  prompt = parsed?.prompt ?? "";
} catch {
  prompt = raw;
}

const results = classify(prompt);
if (!results.length) process.exit(0);

const isFallback = results.length === 1 && results[0]!.label === "_fallback";
const directive = isFallback
  ? FALLBACK.skills
  : "Relevant skills detected. Before acting you MUST invoke the matching skills via the Skill tool " +
    "(process skills like brainstorming/debugging FIRST, then implementation) — don't work from memory. " +
    results.map((r) => `${r.label} -> ${r.skills}`).join("; ") +
    ". Invoke the ones that fit and name which you used.";

process.stdout.write(
  JSON.stringify({
    hookSpecificOutput: {
      hookEventName: "UserPromptSubmit",
      additionalContext: directive,
    },
  }) + "\n"
);

File: adapters/claude-code.py
#!/usr/bin/env python3
"""Claude Code adapter — UserPromptSubmit hook.

Reads the CC hook's stdin JSON ({"prompt": "..."}), classifies it, and
emits the hookSpecificOutput JSON that CC injects as additionalContext.

Wire in ~/.claude/settings.json:
  "UserPromptSubmit": [{"matcher": "", "hooks": [{"type": "command",
    "command": "python3 /path/to/adapters/claude-code.py"}]}]
"""
import json
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).parent.parent))
from core.classify import classify, FALLBACK_HINT

raw = sys.stdin.read()
try:
    prompt = json.loads(raw).get("prompt") or ""
except (ValueError, AttributeError):
    prompt = raw

results = classify(prompt)
if not results:
    sys.exit(0)

if results == [("_fallback", FALLBACK_HINT)]:
    directive = FALLBACK_HINT
else:
    rows = "; ".join(f"{label} -> {skills}" for label, skills in results)
    directive = (
        "Relevant skills detected. Before acting you MUST invoke the matching skills via "
        "the Skill tool (process skills like brainstorming/debugging FIRST, then "
        "implementation) — don't work from memory. "
        + rows
        + ". Invoke the ones that fit and name which you used."
    )

print(json.dumps({
    "hookSpecificOutput": {
        "hookEventName": "UserPromptSubmit",
        "additionalContext": directive,
    }
}))
